### PR TITLE
Feature/1220 bulk ui feedback

### PR DESCRIPTION
**Pull request by richard-jones**
This PR introduced improved feedback on the request for bulk jobs:

* an html (as opposed to alert box) confirmation, with more text and a link to the background jobs page for the job created
* an email notification to the requesting user, when their job completes

It also refactors the javascript on the bulk edit pages for better stability and input validation.

It also introduces the concept of an "ultra" role, which is a user role that not even admins have unless it is explicitly declared (of course, admins can give themselves this roll at any time, it is just a safety mechanism).  This is then applied to the bulk delete code.  This pertains to #1221

### CODE REVIEW

**Comment by Steven-Eardley** [comment]
I've been playing with this locally, and am reliably seeing an error when trying to add a note to 16 applications:

```
bad request {u'status': 400, u'error': u'MapperParsingException[object mapping for [background_job] tried to parse field [reference] as object, but got EOF, has a concrete value been provided to it?]'}
  File "/Users/steve/code/cl/doaj/lib/python2.7/site-packages/werkzeug/serving.py", line 599, in inner
    passthrough_errors, ssl_context).serve_forever()
  File "/Users/steve/code/cl/doaj/lib/python2.7/site-packages/werkzeug/serving.py", line 358, in serve_forever
    HTTPServer.serve_forever(self)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/SocketServer.py", line 238, in serve_forever
    self._handle_request_noblock()
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/SocketServer.py", line 295, in _handle_request_noblock
    self.process_request(request, client_address)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/SocketServer.py", line 321, in process_request
    self.finish_request(request, client_address)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/SocketServer.py", line 334, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/SocketServer.py", line 655, in __init__
    self.handle()
  File "/Users/steve/code/cl/doaj/lib/python2.7/site-packages/werkzeug/serving.py", line 182, in handle
    rv = BaseHTTPRequestHandler.handle(self)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/BaseHTTPServer.py", line 340, in handle
    self.handle_one_request()
  File "/Users/steve/code/cl/doaj/lib/python2.7/site-packages/werkzeug/serving.py", line 217, in handle_one_request
    return self.run_wsgi()
  File "/Users/steve/code/cl/doaj/lib/python2.7/site-packages/werkzeug/serving.py", line 159, in run_wsgi
    execute(app)
  File "/Users/steve/code/cl/doaj/lib/python2.7/site-packages/werkzeug/serving.py", line 148, in execute
    for data in application_iter:
  File "/Users/steve/code/cl/doaj/lib/python2.7/site-packages/werkzeug/debug/__init__.py", line 88, in debug_application
    app_iter = self.app(environ, start_response)
  File "/Users/steve/code/cl/doaj/lib/python2.7/site-packages/flask/app.py", line 1701, in __call__
    return self.wsgi_app(environ, start_response)
  File "/Users/steve/code/cl/doaj/lib/python2.7/site-packages/flask/app.py", line 1687, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/steve/code/cl/doaj/lib/python2.7/site-packages/flask/app.py", line 1358, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/steve/code/cl/doaj/lib/python2.7/site-packages/flask/app.py", line 1344, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Users/steve/code/cl/doaj/lib/python2.7/site-packages/flask_login.py", line 515, in decorated_view
    return fn(*args, **kwargs)
  File "/Users/steve/code/cl/doaj/src/doaj/portality/decorators.py", line 68, in decorated_view
    return fn(*args, **kwargs)
  File "/Users/steve/code/cl/doaj/src/doaj/portality/view/admin.py", line 504, in bulk_add_note
    dry_run=payload.get('dry_run', True)
  File "/Users/steve/code/cl/doaj/src/doaj/portality/tasks/suggestion_bulk_edit.py", line 32, in suggestion_manage
    SuggestionBulkEditBackgroundTask.submit(job)
  File "/Users/steve/code/cl/doaj/src/doaj/portality/tasks/suggestion_bulk_edit.py", line 190, in submit
    background_job.save(blocking=True)
  File "/Users/steve/code/cl/doaj/src/doaj/portality/dao.py", line 131, in save
    traceback.print_stack(file=sys.stdout)
```

**Comment by richard-jones** [comment]
This PR doesn't touch that code, but I suspect this is because during development of the project overall something changed in the mappings.  If you drop your background_jobs index and then start again it should work.

**Comment by Steven-Eardley** [comment]
You're right, dropping the type fixed that. Might need to pay attention to that on the test server, but it may have just been local only.

File: portality/settings.py [hunk 1]
```diff
@@ -145,7 +145,7 @@
 SUPER_USER_ROLE = "admin"
 
 #"api" top-level role is added to all acounts on creation; it can be revoked per account by removal of the role.
-TOP_LEVEL_ROLES = ["admin", "publisher", "editor", "associate_editor", "api"]
+TOP_LEVEL_ROLES = ["admin", "publisher", "editor", "associate_editor", "api", "ultra_bulk_delete"]
```
Comment (emanuil-tolev) [inline portality/settings.py]: mega

**Pull request merged**